theseus/src/state/tags.rs

```diff
@@ -134,71 +134,77 @@ impl Tags {
     // Fetches the tags from the Modrinth API and stores them in the database
     #[tracing::instrument(skip(self))]
     pub async fn fetch_update(&mut self) -> crate::Result<()> {
+        emit_loading(0.1, "starting");
```

Comment:
for state-init related stuff we should probably have the loading in the main init function, which we could use to make a loading screen similar to discord's